DCOS/dcos-testing.sh

```diff
@@ -83,62 +93,73 @@ TEMP_LOGS_DIR="$WORKSPACE/$BUILD_ID"
 VENV_DIR="$WORKSPACE/venv"
 JENKINS_CLI="$WORKSPACE/jenkins-cli.jar"
 
+rm -f $PARAMETERS_FILE && touch $PARAMETERS_FILE && mkdir -p $TEMP_LOGS_DIR && source $UTILS_FILE || exit 1
+
+
+azure_cli_login() {
+    # Validate that credential variables are set
+    if [[ -z $AZURE_SERVICE_PRINCIPAL_ID ]]; then echo "ERROR: Parameter AZURE_SERVICE_PRINCIPAL_ID is not set"; exit 1; fi
+    if [[ -z $AZURE_SERVICE_PRINCIPAL_PASSWORD ]]; then echo "ERROR: Parameter AZURE_SERVICE_PRINCIPAL_PASSWORD is not set"; exit 1; fi
+    if [[ -z $AZURE_SERVICE_PRINCIPAL_TENAT ]]; then echo "ERROR: Parameter AZURE_SERVICE_PRINCIPAL_TENAT is not set"; exit 1; fi
+    # Check if we are already logged in
+    if az account list --output json | jq -r '.[0]["user"]["name"]' | grep -q "^${AZURE_SERVICE_PRINCIPAL_ID}$"; then
+        echo "Account is already logged"
+        return
+    fi
+    # Login
+    az login --output table --service-principal -u $AZURE_SERVICE_PRINCIPAL_ID -p $AZURE_SERVICE_PRINCIPAL_PASSWORD --tenant $AZURE_SERVICE_PRINCIPAL_TENAT
+}
+
+create_linux_ssh_keypair() {
+    echo "Generating a random ssh public/private keypair"
+    ssh-keygen -b 2048 -t rsa -f $LINUX_SSH_KEY_PATH -q -N "" || {
+        echo "ERROR: Failed to generate ssh keypair"
+        return 1
+    }
+    # Convert ssh key to base64 first and remove newlines
+    base64  "$LINUX_SSH_KEY_PATH" | tr -d '\n' >  "${LINUX_SSH_KEY_PATH}.b64"
+    # Upload private/public keys as secrets to Azure key vault
+    az keyvault secret set --vault-name "$AZURE_KEYVAULT_NAME" --name "$LINUX_SSH_KEY_NAME" --file "${LINUX_SSH_KEY_PATH}.b64" &>/dev/null || {
+        echo "ERROR: Failed to upload private key to Azure key vault $AZURE_KEYVAULT_NAME"
+    }
+    az keyvault secret set --vault-name "$AZURE_KEYVAULT_NAME" --name "${LINUX_SSH_KEY_NAME}-pub" --file "${LINUX_SSH_KEY_PATH}.pub" &>/dev/null || {
+        echo "ERROR: Failed to upload private key to Azure key vault $AZURE_KEYVAULT_NAME"
+    }
+    # Export generated public key as variable
+    export LINUX_PUBLIC_SSH_KEY=$(cat ${LINUX_SSH_KEY_PATH}.pub)
+}
+
+generate_windows_password() {
+    echo "Generating random Windows password"
+    WIN_PASSWD="P@s0$(date +%s | sha256sum | base64 | head -c 32)"
+    if [[ -z $WIN_PASSWD ]]; then
+        echo "ERROR: Failed to generate a random Windows password"
+        return 1
+    fi
+    # Upload Windows password to Azure key vault
+    az keyvault secret set --vault-name "$AZURE_KEYVAULT_NAME" --name "$WIN_PASS_SECRET_NAME" --value "$WIN_AGENT_ADMIN_PASSWORD" &>/dev/null || {
```

Comment:
Use `--output table` instead of dropping stdout and stderr.

Reply:
i am redirecting to /dev/null because the default behavior is to print the given value